src/test/utils/overridesHelper.ts

```diff
@@ -1,61 +1,101 @@
 import { Weather, WeatherType } from "#app/data/weather";
 import { Biome } from "#app/enums/biome";
 import * as Overrides from "#app/overrides";
-import { vi } from "vitest";
+import { MockInstance, vi } from "vitest";
+import GameManager from "#test/utils/gameManager";
+import { MysteryEncounterType } from "#enums/mystery-encounter-type";
+import * as overrides from "#app/overrides";
 
 /**
  * Helper to handle overrides in tests
  */
 export class OverridesHelper {
-  constructor() {}
+  game: GameManager;
+  constructor(game: GameManager) {
+    this.game = game;
+  }
 
   /**
    * Override the encounter chance for a mystery encounter.
    * @param percentage the encounter chance in %
+   * @returns spy instance
    */
-  mysteryEncounterChance(percentage: number) {
+  mysteryEncounterChance(percentage: number): MockInstance {
     const maxRate: number = 256; // 100%
     const rate = maxRate * (percentage / 100);
-    vi.spyOn(Overrides, "MYSTERY_ENCOUNTER_RATE_OVERRIDE", "get").mockReturnValue(rate);
+    const spy = vi.spyOn(Overrides, "MYSTERY_ENCOUNTER_RATE_OVERRIDE", "get").mockReturnValue(rate);
     this.log(`Mystery encounter chance set to ${percentage}% (=${rate})!`);
+    return spy;
+  }
+
+  /**
+   * Override the encounter that spawns for the scene
+   * @param encounterType
+   * @returns spy instance
+   */
+  mysteryEncounter(encounterType: MysteryEncounterType): MockInstance {
+    const spy = vi.spyOn(overrides, "MYSTERY_ENCOUNTER_OVERRIDE", "get").mockReturnValue(encounterType);
+    this.log(`Mystery encounter override set to ${encounterType}!`);
+    return spy;
   }
 
   /**
    * Override the starting biome
-   * @warning The biome will not be overridden unless you call `workaround_reInitSceneWithOverrides()` (testUtils)
+   * @warning Any event listeners that are attached to [NewArenaEvent](events\battle-scene.ts) may need to be handled down the line
    * @param biome the biome to set
    */
   startingBiome(biome: Biome) {
-    vi.spyOn(Overrides, "STARTING_BIOME_OVERRIDE", "get").mockReturnValue(biome);
+    this.game.scene.newArena(biome);
     this.log(`Starting biome set to ${Biome[biome]} (=${biome})!`);
   }
 
   /**
    * Override the starting wave (index)
    * @param wave the wave (index) to set. Classic: `1`-`200`
+   * @returns spy instance
    */
-  startingWave(wave: number) {
-    vi.spyOn(Overrides, "STARTING_WAVE_OVERRIDE", "get").mockReturnValue(wave);
+  startingWave(wave: number): MockInstance {
```

Comment:
```suggestion
  startingWave(wave: number) {
```

let it self-inject the type. That is a more specific version of MockInstance